crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -86,42 +101,65 @@ impl Dex {
             .filter_map(future::ready)
     }
 
+    async fn try_solve(
+        &self,
+        order: &Order,
+        dex_order: &dex::Order,
+        tokens: &auction::Tokens,
+        gas_price: auction::GasPrice,
+    ) -> Option<dex::Swap> {
+        self.rate_limiter
+            .execute_with_back_off(
+                async {
+                    let slippage = self.slippage.relative(&dex_order.amount(), tokens);
+                    self.dex
```

Comment:
Would probably also store the result in a variable to avoid having the `match` so far to the right.

Reply:
That actually was the initial approach that produced too much redundant code with returning the same `err` in each case and doing nothing with the result. Extracted the error handler. Let me know if that's still not readable enough.